### ISSUE TITLE
SSL (Let's Encrypt) Configuration Issues 

### ISSUE THREAD

**Issue by DiesDasJenes** [virtualmin/virtualmin-gpl#68]
Aloha,
i updated to the newest version of virtualmin-gpl and requested a new Let's Encrypt certificate. The certificate i had was combined into ssl.combined and ssl.everything. This change was not updated in the virtualmin "Manage SSL Certificate" Page in "Current SSL certificate details". There are still the old links to the ssl.key and ssl.cert. 

How can i fix this linking error?

Operating system | Ubuntu Linux 14.04.4
Virtualmin version | 6.04
Perl version | 5.018002
Apache version | 2.4.7

**Comment by DiesDasJenes** [comment]
Hi, 
sorry by link i meant path to the certificate- or key-file.
My path to the certificate is: "/home/user/ssl.cert"

**Comment by jcameron** [comment]
So you're saying that you requested a new cert from Let's Encrypt using Virtualmin, and even though it appeared to succeed the .cert and .key files weren't updated?

**Comment by DiesDasJenes** [comment]
Yes and the paths in the apache config weren't updated too.

**Comment by DiesDasJenes** [comment]
the path were configured like: "/home/user/ssl.cert"

**Comment by jcameron** [comment]
Ok, so that path shouldn't be changed - however the contents of the file should be updated when you renew the cert, unless there was an error.

**Comment by DiesDasJenes** [comment]
that is what i assumed but not what happened.

**Comment by rostovtsev** [comment]
Are you sure, that Apache was restarted, so new certs could be re-read?

**Comment by DiesDasJenes** [comment]
I am not sure if apache restarted. I didn't checked that and by now i moved away from virtualmin to another server operating system. so i guess we can close this issue.

**Comment by rostovtsev** [comment]
Philipp, this is sad to hear!

Let me ask you, few questions please.

How did you install Virtualmin? Did you just install _.deb_ package or used _install.sh_ script from the official website?

What exactly made you switch? Is it just Let's Encrypt issue or something else?

What server operating system did you switch to?

**Comment by DiesDasJenes** [comment]
I installed virtualmin on a ubuntu 14.04 LTS via the install.sh script from the official website. 

From the start i had issues with virtualmin. The installation failed for me on ubuntu 16.04 without me knowing the reason. I couldn't use certain services and i couldn't figure out what went wrong. Then i moved to ubuntu 14.04 and it worked.

The configuration for hosting a blog or the mail service(DKIM etc.) was complicated with virtualmin and not intuitive. The last thing i tried was creating a sub-virtual server on the frontend to move my blog from my domain to a subdomain. The change resulted in conflicting permissions which i could not resolve by hand or the frontend. In the end Let's encrypt stopped overwriting the certificates. Like i described the certificates were also written to new files but not configured in apache. Also the configuration of the server to use Let's encrypt was also not intuitive and kinda hard. 

The **main** reason i switched was that, when i changed something on the server through virtualmins frontend, i needed to make a backup before because it was most certainly possible that something breaks and i couldn't use certain services anymore.  I wanted to use virtualmin so i don't have to configure every service by hand. In the time i used virtualmin i needed to configure too much by hand. It was good to learn all the configuration stuff through this, but it was not my intention.

I switched now to debain9 and i am using YUNOHOST. 

thanks for your interest.
